### DISCUSSION TITLE
How to allow anyone to mint from a collection?

### DISCUSSION THREAD

**Discussion by aeither** [aptos-labs/aptos-developer-discussions#439] [Questions] [answered]
### Describe your question in detail.

why when i change creator to another user it doesn't work?

```move
        let token_constructor_ref =
            token::create_named_token(
                creator,
                collection_store.collection_name,
                description,
                name,
                option::none(),
                uri
            );
```

![CleanShot 2024-09-17 at 10 13 55@2x](https://github.com/user-attachments/assets/4d7926f6-ee79-4cc9-90fd-31c9ea2ec735)

### What error, if any, are you getting?

public fun address_to_object<T: key>(object: address): Object<T> {
│     │                ----------------- In this function in 0x1::object
│ 193 │         assert!(exists<ObjectCore>(object), error::not_found(EOBJECT_DOES_NOT_EXIST));
│     │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Test was not expected to error, but it aborted with code 393218 originating in the module 0000000000000000000000000000000000000000000000000000000000000001::object rooted here

### What have you tried or looked at?  Or how can we reproduce the error?

create the collection and mint with the same signer works. if used different signer to mint doesn't work

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

### Describe your environment or tooling in detail

Move

**Comment by ajaythxkur** [comment, marked as answer]
@aeither creator should be the collection creator. You can create collection with object signer and move object extend ref to a struct to fetch signer extend ref while minting token later.
